pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -124,6 +124,40 @@ func addLoggerContainerPort(container *v1.Container) {
 	}
 }
 
+func addBatcherAnnotations(batcher *v1alpha2.Batcher, annotations map[string]string) bool {
+	if batcher != nil {
+		annotations[constants.BatcherInternalAnnotationKey] = "true"
+		maxBatchSize := string(*batcher.MaxBatchSize)
```

Comment:
we should check the nil for `MaxBatchSize ` and string() is not the right converter
`s := string(97) // s == "a"`
we should use `s := strconv.Itoa(97) // s == "97"`